alchemy/test/cloudflare/dispatch-namespace.test.ts

```diff
@@ -0,0 +1,229 @@
+      });
+    } finally {
+      await alchemy.destroy(scope);
+      await assertDispatchNamespaceNotExists(dispatchNamespace!.namespace);
+    }
+  });
+
+  test("create worker with dispatch namespace", async (scope) => {
+    const workerName = `${BRANCH_PREFIX}-test-worker-dispatch`;
+    const namespaceName = `claude-main-worker-namespace`;
+
+    let worker: Worker | undefined;
+    let dispatchNamespace: DispatchNamespace | undefined;
+    try {
+      // Create a dispatch namespace
+      dispatchNamespace = await DispatchNamespace("test-dispatch-namespace", {
+        namespace: namespaceName,
+      });
+
+      // Create a worker in the dispatch namespace
+      worker = await Worker(workerName, {
+        name: workerName,
+        script: `
+          export default {
+            async fetch(request, env, ctx) {
+              return new Response('Hello from dispatch namespace!', { status: 200 });
+            }
+          }
+        `,
+        dispatchNamespace: dispatchNamespace,
+        url: false,
+      });
+
+      expect(worker.name).toEqual(workerName);
+      expect(worker.dispatchNamespace).toEqual(dispatchNamespace);
+
+      // Verify worker was deployed to dispatch namespace
+      await assertWorkerInDispatchNamespace(namespaceName, workerName);
+    } finally {
+      await alchemy.destroy(scope);
+      if (worker) {
+        await assertWorkerDoesNotExist(worker.name);
+      }
+      if (dispatchNamespace) {
+        await assertDispatchNamespaceNotExists(dispatchNamespace.namespace);
+      }
+    }
+  });
+
+  test("create worker with dispatch namespace binding", async (scope) => {
+    const workerName = `${BRANCH_PREFIX}-test-worker-dispatch-binding`;
+    const namespaceName = `claude-main-binding-namespace`;
+
+    let worker: Worker | undefined;
+    let dispatchNamespace: DispatchNamespace | undefined;
+    try {
+      // Create a dispatch namespace
+      dispatchNamespace = await DispatchNamespace(
+        "test-dispatch-namespace-binding",
+        {
+          namespace: namespaceName,
+        },
+      );
+
+      // Create a worker with dispatch namespace as binding
+      worker = await Worker(workerName, {
+        name: workerName,
+        script: `
+          export default {
+            async fetch(request, env, ctx) {
+              // NAMESPACE binding should be available
+              return new Response('Hello with dispatch namespace binding!', { status: 200 });
+            }
+          }
+        `,
+        bindings: {
+          NAMESPACE: dispatchNamespace,
+        },
+        url: false,
+      });
+
+      expect(worker.name).toEqual(workerName);
+      expect(worker.bindings.NAMESPACE).toBeDefined();
+    } finally {
+      await alchemy.destroy(scope);
+      if (worker) {
+        await assertWorkerDoesNotExist(worker.name);
+      }
+      if (dispatchNamespace) {
+        await assertDispatchNamespaceNotExists(dispatchNamespace.namespace);
+      }
+    }
+  });
```

Comment:
Just roll these two tests into one test that:
1. creates a dispatch namespace
2. creates a worker in the namespace
3. creates a worker bound to the namespace (dispatcher)
4. executes fetchAndExpectOK against the dispatcher and ensures it works at runtime